drivers/regulator/regulator_max20335.c

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright (c) 2023 Grinn
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT maxim_max20335_regulator
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/dt-bindings/regulator/max20335.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/linear_range.h>
+
+#define MAX20335_BUCK1_CFG        0x0DU
+#define MAX20335_BUCK1_VSET       0x0EU
+#define MAX20335_BUCK2_CFG        0x0FU
+#define MAX20335_BUCK2_VSET       0x10U
+#define MAX20335_BUCK12_CSET      0x11U
+#define MAX20335_BUCK1_CSET_MASK  0xF0U
+#define MAX20335_BUCK2_CSET_MASK  0x0FU
+#define MAX20335_BUCK2_CSET_SHIFT 4
+#define MAX20335_BUCK_EN          BIT(3)
+#define MAX20335_BUCK_EN_MASK     (uint8_t)(BIT(3) | BIT(4))
+
+#define MAX20335_LDO1_CFG         0x12U
+#define MAX20335_LDO1_VSET        0x13U
+#define MAX20335_LDO2_CFG         0x14U
+#define MAX20335_LDO2_VSET        0x15U
+#define MAX20335_LDO3_CFG         0x16U
+#define MAX20335_LDO3_VSET        0x17U
+#define MAX20335_LDO_MODE_MASK    BIT(0)
+#define MAX20335_LDO_EN           BIT(1)
+#define MAX20335_LDO_EN_MASK      (uint8_t)(BIT(1) | BIT(2))
+
+enum max20335_pmic_sources {
+	MAX20335_PMIC_SOURCE_BUCK1,
+	MAX20335_PMIC_SOURCE_BUCK2,
+	MAX20335_PMIC_SOURCE_LDO1,
+	MAX20335_PMIC_SOURCE_LDO2,
+	MAX20335_PMIC_SOURCE_LDO3,
+};
+
+struct regulator_max20335_config {
+	struct regulator_common_config common;
+	struct i2c_dt_spec bus;
+	uint8_t source;
+};
+
+struct regulator_max20335_data {
+	struct regulator_common_data common;
+};
+
+static const struct linear_range buck1_range = LINEAR_RANGE_INIT(700000, 25000U, 0x0U, 0x3FU);
+static const struct linear_range buck2_range = LINEAR_RANGE_INIT(700000, 50000U, 0x0U, 0x3FU);
+static const struct linear_range buck1_current_limit_range =
+	LINEAR_RANGE_INIT(50000, 25000U, 0x02U, 0x0FU);
+static const struct linear_range buck2_current_limit_range =
+	LINEAR_RANGE_INIT(50000, 25000U, 0x02U, 0x0FU);
```

Comment:
```suggestion
static const struct linear_range buck_current_limit_range =
	LINEAR_RANGE_INIT(50000, 25000U, 0x02U, 0x0FU);
```

These are the same, so there only needs to be one.
It will simplify the function too.

Reply:
Right, done.